### ISSUE TITLE
Missing assignment on random shuffle

### ISSUE THREAD

**Issue by janpfeifer** [tianheyu927/PCGrad#5]
First of all: very nice paper! Thanks for writing it up :)

I was reading the code, and I noticed that:

https://github.com/tianheyu927/PCGrad/blob/c5fbd7c856526373828074f06875230f7f3ee79e/PCGrad_tf.py#L36

is missing an assignment (?), I think.

Don't you want to do:

`loss = tf.random.shuffle(loss)`

At least according to the doc here:

https://www.tensorflow.org/api_docs/python/tf/random/shuffle

cheers

**Comment by janpfeifer** [comment]
Ops, I missed the issue has already been brought up ... along with issues of the non-differentiability of `tf.random.shuffle`.

For others landing here:

https://github.com/tianheyu927/PCGrad/issues/3

Closing the issue here.
